### ISSUE TITLE
Blocklists should not block someone you're already following

### ISSUE THREAD

**Issue by meicpalmer** [bluesky-social/social-app#6464]
### Steps to Reproduce

Currently, if you're following a user and that user is added to a blocklist, the blocklist blocks that user.

I've seen posts where people say that they are wary of using blocklists for this reason. I feel like it goes against the intended use of blocklists - to block out random trolls you don't know at all, but someone else spotted them trolling.

Following a user should override any moderation list's attempt to block or mute that user, though a label could still be displayed on their profile saying "this user is on this moderation list".

### What platform(s) does this occur on?

Web (Desktop)

### What version of the app are you using?

1.93.0

**Comment by whetstone303** [comment]
I agree that this is a bug (or more correctly an attack vector). It would also be good if we got warnings if someone we follow is added to a blocklist we're subscribed to; that would provide a lot of good canaries for bad actors weaponizing malicious blocklists.

**Comment by monicaellerose** [comment]
yeah, i would also like this to get added asap! https://github.com/bluesky-social/atproto/discussions/1978#discussioncomment-10616327

**Comment by smenor** [comment]
Just adding a me-to
